src/SwapExtractionTool/VoteExtractionService.cs

```diff
@@ -103,6 +99,62 @@ private async Task ProcessBlockForCollateralVoteTransactionsAsync(BlockTransacti
                 if (opReturnOutput == null)
                     continue;
 
+                // Before checking if it's a vote, check if it's a swap transaction as we now know it has an OP_RETURN.
+                // Ignore any transactions that have inconsequential OP_RETURN values.
+                if (opReturnOutput.Value >= 1.0m)
+                {
+                    // For the purposes of speeding up this search, it doesn't matter per se whether the burn transaction has a valid destination address in it.
+
+                    TransactionModel tx = this.blockExplorerClient.GetTransaction(transaction.TxId);
+
+                    // Check the inputs of the transaction to see if it was funded by one of the vote addresses.
+                    string address = null;
+                    foreach (In input in tx._in)
+                    {
+                        if (input.hash == null)
+                            continue;
+
+                        // The block explorer calls this 'hash' but it is in fact the address that funded the input.
+                        // It is possible that several voting addresses consolidated together in one swap transaction, so just take the first one.
+                        if (this.collateralVotes.ContainsKey(input.hash))
```

Comment:
Kevin, I cant see how this will ever be true as we only ever add to/change the `collateralVotes` array if `address` is not null but then we only set `address` to something if the array contains the `input.hash`?